docs/rest_api.md

```diff
@@ -0,0 +1,558 @@
+
+#### Returns
+ - JSON object containing annotation information about the reset task annotation
+
+
+---
+### `POST /annotations/:typ/:id/merge/:mergedTyp/:mergedId`
+
+Merge two annotations, creating a new explorative.
+
+#### Expects
+ - In the url: `:typ` and `:id` as described above under `GET /annotations/:typ/:id/info`
+ - In the url: `:mergedTyp` and `:mergedId` as described above under `GET /annotations/:typ/:id/info`
+
+
+
+
+---
+### `POST /tasks`
+
+Create tasks without attached NML files
+
+#### Expects
+ - JSON list of objects. Each has these fields:
+   - `"taskTypeId"` `[STRING]`
+   - `"neededExperience"` `[JSON OBJECT]` with these fields:
+     - `"domain"` `[STRING]`
+     - `"value"` `[INT]`
+   - `"openInstances"` `[INT]`
+   - `"projectName"` `[STRING]`
+   - `"scriptId"` (optional) `[STRING]`
+   - `"boundingBox"` (optional) `[JSON OBJECT]` with these fields:
+     - `"topLeft"` `[JSON LIST OF THREE INTS]`
+     - `"width"` `[INT]`
+     - `"height"` `[INT]`
+     - `"depth"` `[INT]`
+   - `"dataSet"` `[STRING]` name of the dataset
+   - `"editPosition"` `[JSON LIST OF THREE INTS]` starting coordinates
+   - `"editRotation"` `[JSON LIST OF THREE INTS]` starting rotation
+   - `"creationInfo"` (optional) `[STRING]` identifier to match with task creation result report
+   - `"description"` (optional) `[STRING]`
+
+
+#### Returns
+
+ - JSON list with status reports about the created tasks, including their ids
+
+#### Note
+
+For each attached NML file, one task is created. This request will also return status code 200 if some tasks could not be created. Failed tasks are instead encoded in the JSON response
+
+
+---
+### `POST /tasks/createFromFiles`
+
+Create tasks with attached NML files
+
+#### Expects
+ - Form data field `"formJSON"`
+   - JSON object with these fields:
+     - `""`
+     - `"taskTypeId"` `[STRING]`
+     - `"neededExperience"` `[JSON OBJECT]` with these fields:
+       - `"domain"` `[STRING]`
+       - `"value"` `[INT]`
+     - `"openInstances"` `[INT]`
+     - `"projectName"` `[STRING]`
+     - `"scriptId"` (optional) `[STRING]`
+     - `"boundingBox"` (optional) `[JSON OBJECT]` with these fields:
+       - `"topLeft"` `[JSON LIST OF THREE INTS]`
+       - `"width"` `[INT]`
+       - `"height"` `[INT]`
+       - `"depth"` `[INT]`
+  - Attached NML files
+
+#### Returns
+
+ - JSON list with status reports about the created tasks, including their ids
+
+#### Note
+
+For each attached NML file, one task is created. This request will also return status code 200 if some tasks could not be created. Failed tasks are instead encoded in the JSON response
+
+
+
+
+---
+### `POST /tasks/list`
+
+List tasks matching search criteria
+
+#### Expects
+ - JSON object with four optional fields:
+   - `"user"` `[STRING]` show only tasks on which the user with this id has worked
+   - `"project"` `[STRING]` show only tasks of the project with this name
+   - `"ids"` `[JSON LIST OF STRINGS]` show only tasks with these ids
```

Comment:
Inconsistent with `LIST[STRING]` used above

Reply:
(changed the one above)